src/debug-generator/debug-generator.c

```diff
@@ -59,12 +62,26 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 if (value) {
                         r = parse_boolean(value);
-                        if (r < 0)
-                                log_error("Failed to parse systemd.debug_shell= argument '%s', ignoring.", value);
-                        else
-                                arg_debug_shell = r;
-                } else
-                        arg_debug_shell = true;
+                        if (r > 0) {
+                                arg_debug_shell = strdup("y");
+                                if (arg_debug_shell == NULL)
+                                        return -ENOMEM;
+                        } else if (r < 0) {
+                                const char *tty;
+
+                                tty = skip_dev_prefix(value);
+                                if (!startswith(tty, "tty"))
+                                        return log_error("Failed to set debug shell argument '%s', ignoring.", value);
+
+                                arg_debug_shell = strdup(tty);
```

Comment:
... and whatever `skip_dev_prefix` returns here without any allocation. It refers to that static argv[] entry so it's OK to use directly.

Reply:
This is not what I'm seeing, unless I'm doing something wrong:

Reply:
OK, so please leave it as it is now.